src/mixins/regions.js

```diff
@@ -72,6 +72,11 @@ export default {
 
     this._regions[name] = region;
 
+    this.listenTo(region, 'destroy', () => {
```

Comment:
I think `listenToOnce` is more complicated even though that's certainly what will happen here.